### PR TITLE
♻️ Make Monster counters `size_t`

### PR DESCRIPTION
**Pull request by glebm**

### CODE REVIEW

File: Source/sync.cpp [hunk 1]
```diff
@@ -232,7 +232,7 @@ bool IsTSyncMonsterValidate(const TSyncMonster &monsterSync)
 {
 	const int monsterId = monsterSync._mndx;
 
-	if (monsterId < 0 || monsterId >= MaxMonsters)
+	if (static_cast<size_t>(monsterId) >= MaxMonsters)
```
Comment (AJenbo) [inline Source/sync.cpp:235]: `_mndx` is already unsinged (`uint8_t`), probably we should just change `monsterId` to `size_t`?

Comment (glebm) [reply Source/sync.cpp]: Done

File: Source/player.cpp [hunk 2]
```diff
@@ -789,7 +789,7 @@ bool PlrHitMonst(int pnum, int monsterId, bool adjacentDamage = false)
 {
 	int hper = 0;
 
-	if (monsterId < 0 || monsterId >= MaxMonsters) {
+	if (static_cast<size_t>(monsterId) >= MaxMonsters) {
```
Comment (AJenbo) [inline Source/player.cpp:792]: All calls to `PlrHitMonst()` verifies that `monsterId` is positive, so we can just make it size_t

File: Source/monster.cpp [hunk 3]
```diff
@@ -348,7 +348,7 @@ void PlaceGroup(int mtype, int num, UniqueMonsterPack uniqueMonsterPack, int lea
 		int y1 = yp;
 
 		if (num + ActiveMonsterCount > totalmonsters) {
-			num = totalmonsters - ActiveMonsterCount;
+			num = static_cast<int>(totalmonsters - ActiveMonsterCount);
```
Comment (AJenbo) [inline Source/monster.cpp:351]: `num`, `totalmonsters`, and `placed` can all be unsigned types

**Comment by AJenbo** [comment]
Looks reasonable :)

File: Source/monster.cpp (hunk 3 above)
Comment (glebm) [reply Source/monster.cpp]: Done

File: Source/player.cpp (hunk 2 above)
Comment (glebm) [reply Source/player.cpp]: Done

Comment (AJenbo) [reply Source/player.cpp]: Didn't expect that you would fully remove the check, but on further inspection it looks safe to do that as well :sweat_smile:

**Review by AJenbo — APPROVED** [review]

**Pull request merged**